packages/office-addin-usage-data/src/usageData.ts

```diff
@@ -272,44 +293,151 @@ export class OfficeAddinUsageData {
   }
 
   /**
-   * Reports custom success event object to Application Insights
-   * @param projectName Project name sent to Application Insights
+   * Reports custom exception event object to Application Insights
+   * @param method Method name sent to Application Insights
+   * @param err Error or message about error sent to Application Insights
    * @param data Data object(s) sent to Application Insights
    */
-  public sendUsageDataSuccessEvent(method: string, data: object = {}) {
+  public reportException(method: string, err: Error | string, data: object = {}) {
+    if (this.getUsageDataLevel() === UsageDataLevel.on) {
+      try {
+        if (err instanceof ExpectedError) {
+          this.reportExpectedException(method, err, data);
+          return;
+        }
+
+        let error = (err instanceof Error) ? err : new Error(`${this.options.projectName} error: ${err}`);
+        error.name = this.getEventName();
+        let exceptionTelemetryObj: appInsights.Contracts.ExceptionTelemetry = {
+          exception: this.maskFilePaths(error),
+          properties: {}
+        };
+        Object.entries({
+          Succeeded: false,
+          Method: method,
+          ExpectedError: false,
+          ...this.defaultData,
+          ...data
+        }).forEach((entry) => {
+          exceptionTelemetryObj.properties[entry[0]] = JSON.stringify(entry[1]);
+        });
+        this.usageDataClient.trackException(exceptionTelemetryObj);
+        this.exceptionsSent++;
+      } catch (e) {
+        this.reportError("sendUsageDataException", e);
+        throw e;
+      }
+    }
+  }
+
+  /**
+   * Reports custom exception event object to Application Insights
+   * @param method Method name sent to Application Insights
+   * @param err Error or message about error sent to Application Insights
+   * @param data Data object(s) sent to Application Insights
+   * @deprecated Use `reportUnexpectedError` instead.  
+   */
+  public sendUsageDataException(method: string, err: Error | string, data: object = {}) {
```

Comment:
This isn't yet in alphabetic order.

Reply:
I have put the non-deprecated functions in alphabetic order, while the deprecated ones just after each one of them.
Do you think putting every function in alphabetic, not caring about deprecated better?

Reply:
Yes, just having them all in alphabetic order. Thanks.